cmd/jujud/agent/machine.go

```diff
@@ -728,14 +730,7 @@ func (a *MachineAgent) startAPIWorkers(apiConn api.Connection) (_ worker.Worker,
 			// the API, report "unknown job type" here.
 		}
 	}
-	if isController {
```

Comment:
The removal of this block and the method it calls are the material changes here.
The rest is drive-by tidy-up.